File: 2076-sum-of-digits-of-string-after-convert/sum-of-digits-of-string-after-convert.java
class Solution {
    public int getLucky(String s, int k) {

        // Convert the string to its numerical form
        StringBuilder sb = new StringBuilder();
        for (char x : s.toCharArray()) {
            sb.append(x - 'a' + 1);
        }

        // Convert the string to the sum of its digits k times
        while (k-- > 0) {
            int sum = 0;
            for (char c : sb.toString().toCharArray()) {
                sum += c - '0';
            }
            sb = new StringBuilder(Integer.toString(sum));
        }

        return Integer.parseInt(sb.toString());
    }
}

File: 2076-sum-of-digits-of-string-after-convert/sum-of-digits-of-string-after-convert.js
/**
 * @param {string} s
 * @param {number} k
 * @return {number}
 */
var getLucky = function(s, k) {
     let numStr = "";
        for (let i = 0; i < s.length; i++) {
            numStr += (s.charCodeAt(i) - 'a'.charCodeAt(0) + 1);
        }

        // Convert the string to the sum of its digits k times
        while (k-- > 0) {
            let sum = 0;
            for (let i = 0; i < numStr.length; i++) {
                sum += parseInt(numStr[i]);
            }
            numStr = sum.toString();
        }

        return parseInt(numStr);
};

 